src/store/chat/index.js

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-console */
+/* eslint-disable arrow-body-style */
+/* eslint-disable prefer-template */
+export default {
+  state: {
+    currentRoom: null,
+    nearbyRooms: [],
+    roomIdToMessages: {}
+  },
+  getters: {
+
+  },
+  mutations: {
+    enterRoom(state, payload) {
+      state.currentRoom = payload.roomId;
+    },
+    leaveRoom(state) {
+      state.currentRoom = null;
+    },
+    updateNearbyRooms(state, payload) {
+      state.nearbyRooms = payload.rooms;
+    },
+    initializeMessages(state, payload) {
+      payload.messages.forEach((message) => {
```

Comment:
do we need a forEach here?

Reply:
Answered from pm: there are other use cases when messages might have different roomIds